questionpy_common/api/attempt.py

```diff
@@ -28,6 +34,36 @@ class CacheControl(Enum):
     NO_CACHE = "NO_CACHE"
 
 
+class QuestionDisplayRole(StrEnum):
+    DEVELOPER = "DEVELOPER"
+    PROCTOR = "PROCTOR"
+    SCORER = "SCORER"
+    TEACHER = "TEACHER"
+
+
+class JsModuleCallRoleFeedback(StrEnum):
+    DEVELOPER = QuestionDisplayRole.DEVELOPER
+    PROCTOR = QuestionDisplayRole.PROCTOR
+    SCORER = QuestionDisplayRole.SCORER
+    TEACHER = QuestionDisplayRole.TEACHER
+    GENERAL_FEEDBACK = "GENERAL_FEEDBACK"
+    SPECIFIC_FEEDBACK = "SPECIFIC_FEEDBACK"
+    RIGHT_ANSWER = "RIGHT_ANSWER"
+    HINT = "HINT"
+
+
+class JsModuleCall(BaseModel, frozen=True):
+    module: str
+    """JS module name like @[package namespace]/[package short name]/[module]"""
+    function: str
+    """Name of a callable value within the JS module."""
+    data: str
+    """JSON data given as argument to the function"""
+    if_role_feedback: JsModuleCallRoleFeedback | None
+    """Function is only called if the user has this role or is allowed to view this feedback type.
+    If None, the function is always called."""
```

Comment:
Irgendwie finde ich es komisch, dass diese beiden Bedingungen durch ein Enum dargestellt werden. Was ist der Gedanke dahinter? Könnte es nicht vorkommen, dass man sowohl die Rolle als auch den Feedback-Typ einschränken will?

Reply:
Könnte schon sinnvoll sein. Ich denke gerade an
```
if_role: QuestionDisplayRole
if_feedback_type: QuestionFeedbackType
```
Und beide Bedingungen sind als UND zu betrachten.

Reply:
Das finde ich besser.

Was ist, wenn eine JS-Funktion abhängig vom Attempt Status aufgerufen werden soll? Zum Beispiel nur, wenn der Versuch noch nicht abgegeben ist?

Reply:
:+1: für die Trennung beider Aspekte.

Das mit dem Attempt-Status könnte das Paket aktuell selbst realisieren:
```python
if not self.scoring_code:
  self.call_js(...)
```

Auf Rollen und Feedback-Optionen erhält das Paket aktuell allerdings keinen Zugriff.
Und ich denke wir sollten sowas auch in JS ermöglichen.